### PR TITLE
Improve JumpDestinationAnalyzer; run in parallel

### PR DESCRIPTION
**Pull request by benaadams**
## Changes

- change to 2 `if`/`else`s rather than 3; shorterns code from 123 bytes of asm to 113 bytes
- Improve code comments (since seems to be discussion point around block gaslimit)
- Remove 2 unused exception classes
- Fix a couple spellings
- Run in parallel on background thread on creation

#### What types of changes does your code introduce?

- [x] Optimization

#### Requires testing

- [x] No

### CODE REVIEW

File: src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs [hunk 1]
```diff
@@ -15,26 +15,28 @@ public sealed class JumpDestinationAnalyzer
...
         private const int BitShiftPerInt32 = 5;
 
-        private int[]? _codeBitmap;
+        private int[]? _jumpDestBitmap;
         public byte[] MachineCode { get; set; }
 
         public JumpDestinationAnalyzer(byte[] code)
         {
+            // Store the code refence as the JumpDest analysis is lazy
+            // and not performed until first jump.
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs:24]: Question - should we start creating bitmap in the background task as soon as we get here? How much would it speed up things? How much not needed work would we do?

**Review by LukaszRozmej — APPROVED** [review]

File: src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs (hunk 1 above)
Comment (benaadams) [reply src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs:24]: Could assume if contract created would be called fairly soon after even if just constructor

Comment (benaadams) [reply src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs:25]: Added

Comment (benaadams) [reply src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs:25]: Yeah, I like that :)

Comment (benaadams) [reply src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs:25]: Yeah, I like that. First use of `IThreadPoolWorkItem` in the repo 😅

Comment (LukaszRozmej) [reply src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs:25]: I would prefer some stats gathering first :)

**Pull request merged**